xprof/btx_timeline.cpp

```diff
@@ -557,6 +560,81 @@ static void memModule_usr_callback(void *btx_handle, void *usr_data, const char
+static perfetto_uuid_t get_nic_parent_track_uuid(
+    timeline_dispatch_t* dispatch,
+    const std::string& hostname,
+    const std::string& ifname) {
+  auto key = std::make_pair(hostname, ifname);
+  auto it  = dispatch->nicParent2uuid.find(key);
+  if (it != dispatch->nicParent2uuid.end())
+    return it->second;
+
+  perfetto_uuid_t parent_uuid = gen_perfetto_uuid();
+  dispatch->nicParent2uuid[key] = parent_uuid;
+
+  auto* packet = dispatch->trace.add_packet();
+  packet->set_trusted_packet_sequence_id(TRUSTED_PACKED_SEQUENCE_ID);
+  packet->set_timestamp(0);  // descriptor at time‐zero
+  packet->set_previous_packet_dropped(true);
+  auto* td = packet->mutable_track_descriptor();
+  td->set_uuid(parent_uuid);
+  td->set_name("NIC Event | Hostname " + hostname + " | " + ifname);
+  td->mutable_counter();     // mark as counter group (so it appears alongside counters)
+  return parent_uuid;
+}
+
+///**
+// * For a given NIC and metric name, generate (or look up) a Perfetto track-UUID
+// * and emit its descriptor.
+// */
+static perfetto_uuid_t get_nic_track_uuid(timeline_dispatch_t *dispatch,
+                                          const std::string &hostname,
+                                          const std::string &ifname,
+                                          const std::string &counter) {
+  // Ensure parent exists
+  perfetto_uuid_t parent_uuid = get_nic_parent_track_uuid(dispatch, hostname, ifname);
+
+  auto key = std::make_tuple(hostname, ifname, counter);
+  auto it  = dispatch->nic2uuid.find(key);
+  if (it != dispatch->nic2uuid.end())
+    return it->second;
+
+  // Create child‐track for this counter
+  perfetto_uuid_t uuid = gen_perfetto_uuid();
+  dispatch->nic2uuid[key] = uuid;
+
+  auto *packet = dispatch->trace.add_packet();
+  packet->set_trusted_packet_sequence_id(TRUSTED_PACKED_SEQUENCE_ID);
+  packet->set_timestamp(0);  // descriptor at time‐zero
+  packet->set_previous_packet_dropped(true);
+  auto *td = packet->mutable_track_descriptor();
+  td->set_uuid(uuid);
+  td->set_parent_uuid(parent_uuid);
+  td->set_name(counter);
+  td->mutable_counter();
+  return uuid;
+}
+
+static void nic_usr_callback(void*       btx_handle,
+                             void*       usr_data,
+                             const char* hostname,
+                             int64_t     ts,
+                             const char* ifname,
+                             const char* counter,
+                             uint64_t    value) {
+  auto* dispatch = static_cast<timeline_dispatch_t*>(usr_data);
+  perfetto_uuid_t uuid = get_nic_track_uuid(dispatch, hostname, ifname, counter);
+
+  auto* packet = dispatch->trace.add_packet();
+  packet->set_trusted_packet_sequence_id(TRUSTED_PACKED_SEQUENCE_ID);
+  packet->set_timestamp(ts);
+  auto* ev = packet->mutable_track_event();
+  ev->set_type(perfetto_pruned::TrackEvent::TYPE_COUNTER);
+  ev->set_track_uuid(uuid);
+  ev->set_double_counter_value(static_cast<double>(value));
```

Comment:
so you print the value, but it's now a `diff` no?
 It's just an offset  / renormilization , I see! 
 
 I think will be cleaner to just no do any shenaningan, and print the raw value. Will make it easier to debug

Reply:
If level zero do the re-scaling to 0 into the  is daemon let do it there. I guess it will be better to do into the `timeline` as potential other UI may do the rescaling when visualizing